src/Maker/MakeTest.php

```diff
@@ -0,0 +1,189 @@
+        'ApiTestCase' => 'https://api-platform.com/docs/distribution/testing/',
+        'PantherTestCase' => 'https://github.com/symfony/panther#testing-usage',
+    ];
+
+    public static function getCommandName(): string
+    {
+        return 'make:test';
+    }
+
+    /**
+     * @deprecated remove this method when removing make:unit-test and make:functional-test
+     */
+    public static function getCommandAliases(): iterable
+    {
+        yield 'make:unit-test';
+        yield 'make:functional-test';
+    }
+
+    public static function getCommandDescription(): string
+    {
+        return 'Creates a new test class';
+    }
+
+    public function configureCommand(Command $command, InputConfiguration $inputConf)
+    {
+        $typesDesc = [];
+        $typesHelp = [];
+        foreach (self::DESCRIPTIONS as $type => $desc) {
+            $typesDesc[] = sprintf('<fg=yellow>%s</> (%s)', $type, $desc);
+            $typesHelp[] = sprintf('* <info>%s</info>: %s', $type, $desc);
+        }
+
+        $command
+            ->addArgument('name', InputArgument::OPTIONAL, 'The name of the test class (e.g. <fg=yellow>BlogPostTest</>)')
+            ->addArgument('type', InputArgument::OPTIONAL, 'The type of test: '.implode(', ', $typesDesc))
+            ->setHelp(file_get_contents(__DIR__.'/../Resources/help/MakeTest.txt').implode("\n", $typesHelp));
+
+        $inputConf->setArgumentAsNonInteractive('type');
+    }
+
+    public function interact(InputInterface $input, ConsoleStyle $io, Command $command)
+    {
+        /* @deprecated remove the following block when removing make:unit-test and make:functional-test */
+        $currentCommand = $input->getFirstArgument();
+        switch ($currentCommand) {
+            case 'make:unit-test':
+                $input->setArgument('type', 'TestCase');
+                $io->caution('The "make:unit-test" command is deprecated, use "make:test" instead.');
+
+                return;
+
+            case 'make:functional-test':
+                $input->setArgument('type', trait_exists(PantherTestCaseTrait::class) ? 'WebTestCase' : 'PantherTestCase');
+                $io->caution('The "make:functional-test" command is deprecated, use "make:test" instead.');
+
+                return;
+        }
+
+        if (null !== $type = $input->getArgument('type')) {
+            if (!isset(self::DESCRIPTIONS[$type])) {
+                throw new RuntimeCommandException(sprintf('The test type must be one of "%s", "%s" given.', implode('", "', array_keys(self::DESCRIPTIONS)), $type));
+            }
+
+            return;
+        }
+
+        $input->setArgument(
+            'type',
+            $io->choice('Which test type would you like?', self::DESCRIPTIONS)
+        );
+    }
+
+    public function generate(InputInterface $input, ConsoleStyle $io, Generator $generator)
+    {
+        $testClassNameDetails = $generator->createClassNameDetails(
+            $input->getArgument('name'),
```

Comment:
Do we ever ask this interactively?

Reply:
Yes, it's delegated to the method in the abstract class.